### PR TITLE
Posture Management SDK

### PR DESCRIPTION
**Pull request by jasoncalalang**
This is for creating the SDK and its corresponding integration tests for Security and Compliance Center

### CODE REVIEW

File: notificationsv1/notifications_v1_test.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ import (
 	. "github.com/onsi/gomega"
 )
 
-var _ = Describe(`NotificationsV1`, func() {
+var _ = XDescribe(`NotificationsV1`, func() {
```
Comment (ashishth09) [inline notificationsv1/notifications_v1_test.go:37]: I hope this is temporary to just see the posture code running and you will remove it eventually. Same for governance

File: examples/posturemanagementv1/CreateCollector.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package examples
```
Comment (ashishth09) [inline examples/posturemanagementv1/CreateCollector.go:1]: Examples will be auto generated on passing --genExamples and will appear inside the posturemanagementv1 folder. For example https://github.com/IBM/scc-go-sdk/blob/main/findingsv1/findings_v1_examples_test.go

File: posturemanagementv1/posture_management_v1_integration_test.go [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+			Expect(statusCode).To(Equal(201))
+			Expect(id).NotTo(BeNil())
+		})
+
+		//new credential to be created
+		It(`Create Credential`, func() {
+			fmt.Println(`Create Credential`)
+			var statusCode int
+			credentialPath := config["CREDENTIAL_PATH"]
+			pemPath := config["PEM_PATH"]
```
Comment (ashishth09) [inline posturemanagementv1/posture_management_v1_integration_test.go:119]: I assume the PEM_PATH and CREDENTIAL_PATH are exposed as 

POSTURE_MANAGEMENT_PEM_PATH and  POSTURE_MANAGEMENT_CREDENTIAL_PATH in the value of POSTURE_MANAGEMENT_ENV set in travis. Then we don't need

```
	echo "${POSTURE_PEM_ENV}" | base64 -d >> posture-credential.pem
       echo "${POSTURE_CREDENTIAL_ENV}" | base64 -d >> posture-credential.json
```
in build/testScript.sh

File: testInput/json/create-credentials.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+	"credential_name": "SDKDEMO CREATE CREDENTIAL",
+	"credential_description": "SDK DEMO JASON TEST",
+	"purpose": "discovery_fact_collection_remediation",
+	"enabled": true,
+	"credential_group": {},
+	"credential_type": "ibm_cloud",
+	"ibm_cloud": {
+		"ibm_api_key": "yG7CpPf0fsClxZN3In0sh24SWC6AQgIzm9kq4hH9j5_7"
```
Comment (ashishth09) [inline testInput/json/create-credentials.json:9]: Please remove this and delete the api key also.

Comment (ashishth09) [reply testInput/json/create-credentials.json:9]: API Key should be read from the travis envs

File: testInput/json/key.pem [hunk 5]
```diff
@@ -0,0 +1,28 @@
+-----BEGIN PRIVATE KEY-----
```
Comment (ashishth09) [inline testInput/json/key.pem:1]: A candidate for env as well.

**Review by ashishth09 — CHANGES_REQUESTED** [review]

**Comment by ashishth09** [comment]
Are integration test hand coded or auto generated? Looks like the former.

**Comment by blinxguna** [comment]
> Are integration test hand coded or auto generated? Looks like the former.

@ashishth09 they are not auto generated. we have written the test case.

File: testInput/json/create-credentials.json (hunk 4 above)
Comment (blinxguna) [reply testInput/json/create-credentials.json:9]: Yes we can remove now. we move then to base64 encoded

File: testInput/json/key.pem (hunk 5 above)
Comment (blinxguna) [reply testInput/json/key.pem:1]: Yes we can remove now. we move then to base64 encoded

File: posturemanagementv1/posture_management_v1_integration_test.go (hunk 3 above)
Comment (jasoncalalang) [reply posturemanagementv1/posture_management_v1_integration_test.go:119]: @ashishth09 the CREDENTIAL and PEM are the file contents. The create credentials API needs to read from those files and sends them as multipart/form-data. That is why we need the path and contents.

File: notificationsv1/notifications_v1_test.go (hunk 1 above)
Comment (blinxguna) [reply notificationsv1/notifications_v1_test.go]: Fixed

File: go.mod [hunk 6]
```diff
@@ -1,4 +1,4 @@
-module github.com/ibm/scc-go-sdk
+module github.com/ibm-cloud-security/scc-go-sdk
```
Comment (gary1998) [inline go.mod:1]: you shouldn't change the module name of any go module... it creates many problems

**Comment by ashishth09** [comment]
> > Are integration test hand coded or auto generated? Looks like the former.
> 
> @ashishth09 they are not auto generated. we have written the test case.

Did you try generating it? Any issues in generated test cases? We are using the generated ones and they seem to cover the tests cases at functional level like actually calling the APIs.

File: posturemanagementv1/posture_management_v1_integration_test.go (hunk 3 above)
Comment (ashishth09) [reply posturemanagementv1/posture_management_v1_integration_test.go:119]: Where are these files read in the tests cases posture-credential.pem and posture-credential.json? I couldn't find them.

**Comment by jasoncalalang** [comment]
> > > Are integration test hand coded or auto generated? Looks like the former.
> > 
> > 
> > @ashishth09 they are not auto generated. we have written the test case.
> 
> Did you try generating it? Any issues in generated test cases? We are using the generated ones and they seem to cover the tests cases at functional level like actually calling the APIs.

generated and updated

File: go.mod (hunk 6 above)
Comment (jasoncalalang) [reply go.mod]: previously this was under `ibm-cloud-security`, it was changed after we merged from main

File: posturemanagementv1/posture_management_v1_integration_test.go (hunk 3 above)
Comment (jasoncalalang) [reply posturemanagementv1/posture_management_v1_integration_test.go:123]: > Where are these files read in the tests cases posture-credential.pem and posture-credential.json? I couldn't find them.

those files are generated in `testScript.sh`

File: examples/posturemanagementv1/CreateCollector.go (hunk 2 above)
Comment (gary1998) [reply examples/posturemanagementv1/CreateCollector.go:1]: examples have to be generated only, we don't need manual examples... please delete this after you've generated examples

File: posturemanagementv1/posture_management_v1_integration_test.go (hunk 3 above)
Comment (gary1998) [reply posturemanagementv1/posture_management_v1_integration_test.go:123]: you cannot use external files or examples in ITs.... just fix test cases that are there in generated ITs and if required you can add more test cases.

**Review by gary1998 — CHANGES_REQUESTED** [review]
Also, please try testing your ITs locally first.... travis currently doesn't run ITs on CI.... we're moving to Github Actions that will fix the problem.

File: examples/posturemanagementv1/CreateCollector.go (hunk 2 above)
Comment (jasoncalalang) [reply examples/posturemanagementv1/CreateCollector.go]: accepted and updated

File: posturemanagementv1/posture_management_v1_integration_test.go (hunk 3 above)
Comment (jasoncalalang) [reply posturemanagementv1/posture_management_v1_integration_test.go]: accepted and will update accordingly

**Comment by jasoncalalang** [comment]
> Are integration test hand coded or auto generated? Looks like the former.

Yes you are correct, but we will generate the ITs as advised, and update this PR accordingly.

**Comment by gary1998** [comment]
Just took a quick glance to PostureMgmt API doc, and found that `AccountID` isn't used as global parameter in the SDK service. You'll have to add `x-sdk-global-param` to `AccountId` component in the API doc, as in https://github.ibm.com/project-fortress/scc-sdk-definitions/blob/master/findings.yaml#L772

This regeneration will effect all files within the service, so please make sure UTs are fine, and also try running ITs locally.

**Comment by jasoncalalang** [comment]
> Just took a quick glance to PostureMgmt API doc, and found that `AccountID` isn't used as global parameter in the SDK service. You'll have to add `x-sdk-global-param` to `AccountId` component in the API doc, as in https://github.ibm.com/project-fortress/scc-sdk-definitions/blob/master/findings.yaml#L772
> 
> This regeneration will effect all files within the service, so please make sure UTs are fine, and also try running ITs locally.

Accepted, will make the necessary change in the YAML as well, to re-generate the SDK stubs.

**Comment by jasoncalalang** [comment]
will close this PR for now, as discussed with @gary1998

**Pull request closed without merging**